keras/engine/training.py

```diff
@@ -2005,15 +2009,33 @@ def generate_arrays_from_file(path):
         if do_validation:
             self._make_test_function()
 
+        is_sequence = isinstance(generator, Sequence)
+        if not is_sequence and use_multiprocessing and workers > 1:
```

Comment:
This code is moved up since we need to resolve `steps_per_epoch` before callback parameters are instantiated.